packages/integration-sdk-runtime/src/synchronization/index.ts

```diff
@@ -260,6 +261,53 @@ interface UploadDataChunkParams<T extends UploadDataLookup, K extends keyof T> {
   batch: T[K][];
 }
 
+type HandleUploadDataChunkErrorParams = {
+  err: any;
+  attemptContext: AttemptContext;
+  logger: IntegrationLogger;
+};
+
+function handleUploadDataChunkError({
+  err,
+  attemptContext,
+  logger,
+}: HandleUploadDataChunkErrorParams): void {
+  if (err.code === 'RequestEntityTooLargeException') {
+    // No reason to retry these errors as the request size ain't gonna change.
+    throw new IntegrationError({
+      code: 'INTEGRATION_UPLOAD_FAILED',
+      cause: err,
+      fatal: false,
```

Comment:
This is non-fatal because other steps should still run if one step fails to upload